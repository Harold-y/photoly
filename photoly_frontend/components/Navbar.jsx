import {Button, Heading, HStack, Image, Avatar} from "@chakra-ui/react";

export default function Navbar() {
    return (
        <HStack bg={"teal.400"} w={"100%"} h={16} color={"white"} p={4}
                justifyContent={"space-between"}>
            <Heading ml={4}>PHOTOLY LOGO</Heading>
            <HStack h={"inherit"} p={2} spacing={4} mr={4}>
                <Button variant={"link"} color={"white"}>Logout</Button>
                <Button variant={"link"} color={"white"}>Manage</Button>
                <Avatar
                    borderRadius='full'
                    maxH={"100%"}
                    maxW={"100%"}
                    src='https://bit.ly/dan-abramov'
                    alt='Avatar'
                />
            </HStack>
        </HStack>
    )

}
