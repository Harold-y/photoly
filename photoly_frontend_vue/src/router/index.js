import { createRouter, createWebHistory } from 'vue-router'
import Welcome from '../views/Welcome.vue'
import Login from '../views/Login.vue'
import Signup from '../views/Signup.vue'
import Home from '../views/Home.vue'
import MoveToShowFolderVue from '@/components/MoveToShowFolder.vue'
import AdminDashboard from '@/views/AdminDashboard.vue'
import AdminUserManagement from '@/views/AdminUserManagement.vue'

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes: [
    {
      path: "/",
      name: "welcome",
      component: Welcome,
   },
   {
    path: "/login",
    name: "login",
    component: Login,
   },
   {
    path: "/signup",
    name: "signup",
    component: Signup,
   },
   {
    path: "/home/:part",
    name: "home",
    component: Home,
    props: true
   },
   {
    path: "/test",
    name: "test",
    component: MoveToShowFolderVue,
   },
   {
    path: "/thepersonincharge/dashboard",
    name: "thepersonincharge_dashboard",
    component: AdminDashboard,
   },
   {
    path: "/thepersonincharge/usermanage",
    name: "thepersonincharge_usermanage",
    component: AdminUserManagement,
   }
  ]
})

export default router
