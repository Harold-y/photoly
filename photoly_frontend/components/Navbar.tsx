import {Button, Text, HStack, Avatar} from "@chakra-ui/react";
import {useRouter} from "next/router";
import Image from 'next/image'
import React from "react";
import useLocalStorage, {TOKEN_KEY} from "../hooks/useLocalStorage";

const Navbar: React.FC = () => {
    const router = useRouter()
    const {remove} = useLocalStorage(TOKEN_KEY)
    const handleLogout = () => {
        router.push("/login")
        remove()
    }
    return (
        <HStack bg={"teal.400"} w={"100%"} h={16} color={"white"} p={4}
                justifyContent={"space-between"}>
            <HStack>
                <Image src={"/favicon.ico"} alt={"logo"} width={"50px"} height={"50px"} data-testid={"logo"}/>
                <Text fontSize={"xl"} fontWeight={"medium"}>PHOTOLY</Text>
            </HStack>
            <HStack h={"inherit"} p={2} spacing={4} mr={4}>
                <Button variant={"link"} color={"white"} onClick={handleLogout}>Logout</Button>
                <Button variant={"link"} color={"white"} onClick={() => router.push("/manage")}>Manage</Button>
                <Avatar
                    borderRadius='full'
                    maxH={"100%"}
                    maxW={"100%"}
                    src='https://bit.ly/dan-abramov'
                    data-testid={"avatar"}
                />
            </HStack>
        </HStack>
    )
}
export default Navbar
