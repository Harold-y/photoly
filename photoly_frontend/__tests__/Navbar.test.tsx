import React from "react";
import {getByTestId, render***REMOVED*** from "@testing-library/react"
import { useRouter ***REMOVED*** from 'next/router'

import Navbar from "../components/Navbar";
import {describe, expect***REMOVED*** from "@jest/globals";
import userEvent from "@testing-library/user-event";

// useRouter now returns a mock function in this test file
jest.mock('next/router', () => ({
    __esModule: true,
    useRouter: jest.fn(() => {return {
        push: jest.fn() // the component uses `router.push` only
***REMOVED******REMOVED***),
***REMOVED***))

describe("<Navbar/> rendering", () => {
    test("should render navigation bar on the screen", async () => {
        const {container, getByText, getByTestId***REMOVED*** = render(<Navbar/>);
        expect(container.firstChild).toMatchSnapshot()
        expect(getByText("PHOTOLY")).toBeTruthy()
        expect(getByText("Logout")).toBeTruthy()
        expect(getByText("Manage")).toBeTruthy()
        expect(getByTestId("logo")).toBeTruthy()
        expect(getByTestId("avatar")).toBeTruthy()
***REMOVED***)
***REMOVED***)


describe("<Navbar/> functionality", () => {
    test("test manage page redirection", async () => {
        const mockRouter = {
            push: jest.fn() // the component uses `router.push` only
***REMOVED***;
        // make useRouter return an object with a fake push function in it
        (useRouter as jest.Mock).mockReturnValue(mockRouter)
        const {getByText***REMOVED*** = render(<Navbar/>);

        await userEvent.click(getByText(/manage/i))
        expect(mockRouter.push).toHaveBeenCalledWith("/manage")
***REMOVED***)

    test("test logout page redirection", async () => {
        const mockRouter = {
            push: jest.fn() // the component uses `router.push` only
***REMOVED***;
        // make useRouter return an object with a fake push function in it
        (useRouter as jest.Mock).mockReturnValue(mockRouter)
        const {getByText***REMOVED*** = render(<Navbar/>);

        await userEvent.click(getByText(/logout/i))
        expect(mockRouter.push).toHaveBeenCalledWith("/login")
***REMOVED***)
***REMOVED***)