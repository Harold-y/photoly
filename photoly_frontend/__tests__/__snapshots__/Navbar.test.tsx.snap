// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> rendering should render navigation bar on the screen 1`] = `
<div
  class="chakra-stack css-1a1oqyw"
>
  <div
    class="chakra-stack css-84zodg"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2750%27%20height=%2750%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="logo"
        data-nimg="intrinsic"
        data-testid="logo"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
    <p
      class="chakra-text css-ahuls7"
    >
      PHOTOLY
    </p>
  </div>
  <div
    class="chakra-stack css-1y4bou5"
  >
    <button
      class="chakra-button css-117sk0y"
      type="button"
    >
      Logout
    </button>
    <button
      class="chakra-button css-117sk0y"
      type="button"
    >
      Manage
    </button>
    <span
      class="chakra-avatar css-175y0vd"
      data-testid="avatar"
    >
      <svg
        aria-label=" avatar"
        class="chakra-avatar__svg css-16ite8i"
        role="img"
        viewBox="0 0 128 128"
      >
        <path
          d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
          fill="currentColor"
        />
        <path
          d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
          fill="currentColor"
        />
      </svg>
    </span>
  </div>
</div>
`;
